scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,249 @@
+        if (( $# > 1 ));
+        then { arg_output="$2"; shift; }
+        else { echo "$0: missing output filename" >&2; exit 1; }
+        fi;;
+    (--silent|-s)
+        arg_verbose=false;;
+    (-X)
+        set -x;;
+    (-h|--help)
+        display_help
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/KUBECONFIG="$in_KUBECONFIG" kubectl/generated/KUBECONFIG="$in_KUBECONFIG" kubectl_config/
+echov "Determining the list of kubeconfigs to search:"
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    in_KUBECONFIG="$arg_kubeconfig"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$arg_kubeconfig"
+    fi
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    in_KUBECONFIG="$KUBECONFIG"
+    if [[ "$arg_output" == "" ]] ; then
+        if [[ "$KUBECONFIG" == *":"* ]] ; then
+            # hard to decide
+            # out_kubeconfig="$HOME/.kube/config"
+            >&2 echo "ERROR: please soecify an output kubeconfig filename with \"-o <filename>\"!"
+            exit 1
+        else
+            out_kubeconfig="$KUBECONFIG"
+        fi
+    fi
+else
+    in_KUBECONFIG="$HOME/.kube/config"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$HOME/.kube/config"
+    else
+        out_kubeconfig="$arg_output"
+    fi
+fi
+echov "- input kubeconfigs: \"$in_KUBECONFIG\""
+echov "- output kubeconfig: \"$out_kubeconfig\""
+
+
+# Determine the list of contexts to search
+echov "Determining the list of contexts to search:"
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(KUBECONFIG="$in_KUBECONFIG" kubectl config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+for context in "${contexts[@]}" ; do # for all contexts
+    echov "- \"${context}\""
+done
+
+
+echov "Getting kubeconfigs of KubeFlex Control Planes:"
+n=0
+for context in "${contexts[@]}" ; do # for all contexts
+    echov "- searching context \"${context}\""
+    cps=($(KUBECONFIG="$in_KUBECONFIG" kubectl --context "${context}" get controlplanes -no-headers -o name 2> /dev/null))
+    for i in "${!cps[@]}" ; do # for all control planes in context ${context}
+        name=${cps[i]##*/}
+        if [[ "$arg_names" != "" && ",$arg_names," != *",$name,"* ]] ; then
+            echov "  - skipping \"$name\""
+            continue
+        fi
+        cp_name[n]=$name
+        cp_type[n]=$(KUBECONFIG="$in_KUBECONFIG" kubectl get controlplane ${cp_name[n]} -o jsonpath='{.spec.type}')
+        echov "  - found \"${cp_name[i]}\" of type \"${cp_type[i]}\""
+        cp_kubeconfig[n]=$(get_kubeconfig "${context}" "${cp_name[n]}" "${cp_type[n]}" "$arg_ip_addr")
```

Comment:
When I tested this, it assigned to `cp_kubeconfig[n]` the output from line 47 (`    echov "Getting the kubeconfig of Control Plane \"$cp_name\" of type \"$cp_type\" in context \"$context\":"`).

Reply:
I solved this issue by changing echov to print on stderr